client/my-sites/plugins/plugins-list/index.jsx

```diff
@@ -131,12 +131,12 @@ export default React.createClass( {
 		const bulkManagement = ! this.state.bulkManagement;
 
 		if ( bulkManagement ) {
+			this.setBulkSelectionState( this.props.plugins, bulkManagement );
 			this.setState( { bulkManagement } );
 			return this.recordEvent( 'Clicked Manage' );
 		}
 
-		// Unselect all plugins.
-		this.setState( { selectedPlugins: {}, bulkManagement } );
```

Comment:
I removed this for clarity. The default state should be enabling all the plugins (via `setBulkSelectionState`) and not empty. 

So even if the user changes the state of selections we don't need to clear those changes out when toggling off  The default state will now be set when toggling on.